src/jmhTest/java/io/deephaven/csv/benchmark/intcol/IntColumnBenchmarkTest.java

```diff
@@ -0,0 +1,63 @@
+package io.deephaven.csv.benchmark.intcol;
+
+import io.deephaven.csv.benchmark.intcol.IntColumnBenchmark.InputProvider;
+import io.deephaven.csv.benchmark.intcol.IntColumnBenchmark.ReusableStorage;
+import io.deephaven.csv.benchmark.util.BenchmarkResult;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class IntColumnBenchmarkTest {
+
+    IntColumnBenchmark benchmark;
+    InputProvider inputProvider;
+    ReusableStorage reusableStorage;
+    BenchmarkResult<int[]> result;
+
+    @BeforeEach
+    void setUp() {
+        benchmark = new IntColumnBenchmark();
+        inputProvider = new InputProvider();
+        reusableStorage = new ReusableStorage();
+    }
+
+    @AfterEach
+    void tearDown() {
+        inputProvider.check(result);
+    }
+
+    @Test
+    void deephaven() throws Exception {
+        result = benchmark.deephaven(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void apache() throws Exception {
+        result = benchmark.apache(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void fastCsv() throws Exception {
+        result = benchmark.fastCsv(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void jackson() throws Exception {
+        result = benchmark.jackson(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void openCsv() throws Exception {
+        result = benchmark.openCsv(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void simpleFlatMapper() throws Exception {
+        result = benchmark.simpleFlatMapper(inputProvider, reusableStorage);
+    }
+
+    @Test
+    void superCsv() throws Exception {
+        result = benchmark.superCsv(inputProvider, reusableStorage);
+    }
+}
```

Comment:
I think it's super great to have tests on the benchmarks (so, thank you) but I think this style of code misses out on an opportunity to write this all in functional, side-effect-free style. Put another way I don't much care for the instance methods and the spooky-action-from-a-distance of `@BeforeEach` and `@AfterEach`. I think it might be nicer to write it like this, with a little helper method. What do you think?

```
public class IntColumnBenchmarkTest {
    @Test
    void deephaven() throws Exception {
        runTest(new IntColumnBenchmark()::deephaven);
    }

    @Test
    void apache() throws Exception {
        runTest(new IntColumnBenchmark()::apache);
    }

    @Test
    void fastCsv() throws Exception {
        runTest(new IntColumnBenchmark()::fastCsv);
    }

    @Test
    void jackson() throws Exception {
        runTest(new IntColumnBenchmark()::jackson);
    }

    @Test
    void openCsv() throws Exception {
        runTest(new IntColumnBenchmark()::openCsv);
    }

    @Test
    void simpleFlatMapper() throws Exception {
        runTest(new IntColumnBenchmark()::simpleFlatMapper);
    }

    @Test
    void superCsv() throws Exception {
        runTest(new IntColumnBenchmark()::superCsv);
    }

    private static void runTest(MyFunction<InputProvider, ReusableStorage, BenchmarkResult<int[]>> benchmark) throws Exception {
        final InputProvider inputProvider = new InputProvider();
        final ReusableStorage reusableStorage = new ReusableStorage();
        final BenchmarkResult<int[]> result = benchmark.apply(inputProvider, reusableStorage);
        inputProvider.check(result);
    }

    private interface MyFunction<T, U, R> {
        R apply(T t, U u) throws Exception;
    }
}
```

Reply:
I do like my checks in the actual test body instead of `@AfterEach`, and there is an opportunity to be more functional, but I think there is value in using well-worn junit scaffolding for state management with setup/BeforeEach. I've pushed changes in these regards, let me know if you think it's acceptable.